### PR TITLE
ansible-galaxy - support `resolvelib >= 0.5.3, < 1.1.0`

### PR DESCRIPTION
**Pull request by hswong3i**
##### SUMMARY
<https://pypi.org/project/resolvelib/1.0.1> released on 2023-03-09:

-   <https://github.com/sarugaku/resolvelib/blob/main/CHANGELOG.rst#101-2023-03-09>
-   <https://github.com/sarugaku/resolvelib/releases/tag/1.0.1>

##### ISSUE TYPE
- Feature Pull Request

##### ADDITIONAL INFORMATION

API guarantees discussions:
* https://github.com/sarugaku/resolvelib/issues/97
* https://github.com/sarugaku/resolvelib/pull/125

### CODE REVIEW

**Review by webknjaz — APPROVED** [review]

File: test/lib/ansible_test/_data/requirements/ansible.txt [hunk 1]
```diff
@@ -15,4 +15,4 @@ importlib_resources >= 5.0, < 5.1; python_version < '3.10'
 # NOTE: Ref: https://github.com/sarugaku/resolvelib/issues/69
 # NOTE: When updating the upper bound, also update the latest version used
 # NOTE: in the ansible-galaxy-collection test suite.
-resolvelib >= 0.5.3, < 0.10.0  # dependency resolver used by ansible-galaxy
+resolvelib >= 0.5.3, < 1.1.0  # dependency resolver used by ansible-galaxy
```
Comment (mattclay) [inline test/lib/ansible_test/_data/requirements/ansible.txt:18]: To verify the new version works, temporarily change this to `resolvelib==1.0.1`.

File: requirements.txt [hunk 2]
```diff
@@ -15,4 +15,4 @@ importlib_resources >= 5.0, < 5.1; python_version < '3.10'
 # NOTE: Ref: https://github.com/sarugaku/resolvelib/issues/69
 # NOTE: When updating the upper bound, also update the latest version used
 # NOTE: in the ansible-galaxy-collection test suite.
-resolvelib >= 0.5.3, < 0.10.0  # dependency resolver used by ansible-galaxy
+resolvelib >= 0.5.3, < 1.1.0  # dependency resolver used by ansible-galaxy
```
Comment (mattclay) [inline requirements.txt:18]: To verify the new version works, temporarily change this to `resolvelib==1.0.1`.

**Review by mattclay — CHANGES_REQUESTED** [review]
Temporary changes are needed to verify the new version passes in CI. Once it passes, those changes can be reverted.

File: requirements.txt (hunk 2 above)
Comment (webknjaz) [reply requirements.txt:18]: I'd rather extend https://github.com/ansible/ansible/blob/bad8843/test/integration/targets/ansible-galaxy-collection/vars/main.yml#L7-L11 with more versions.

Comment (mattclay) [reply requirements.txt:18]: That should be updated as well.

@webknjaz Do you know if that covers all of our uses of resolvelib? If not, the temporary commit and revert would still be useful.

Comment (hswong3i) [reply requirements.txt:18]: also update for https://github.com/ansible/ansible/blob/bad8843/test/integration/targets/ansible-galaxy-collection/vars/main.yml#L7-L11

Comment (webknjaz) [reply requirements.txt:18]: @mattclay I don't know. It's best to ask @s-hertel.

Comment (mattclay) [reply requirements.txt:18]: @hswong3i Tests passed with the pinned version. Go ahead and revert that commit. No need to force push the revert, we'll just squash the commits when merging the PR.

Comment (hswong3i) [reply requirements.txt:18]: done ;-)

**Review by mattclay — APPROVED** [review]

**Review by s-hertel — APPROVED** [review]

**Pull request merged**